scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/Record.java

```diff
@@ -750,13 +814,18 @@ public long getSequenceNumber() {
 	 * @param sequenceNumber the new sequence number
 	 * @throws GeneralSecurityException if the fragment could not be re-encrypted
 	 */
-	public synchronized void setSequenceNumber(long sequenceNumber) throws GeneralSecurityException {
+	public void setSequenceNumber(long sequenceNumber) throws GeneralSecurityException {
 		if (sequenceNumber > MAX_SEQUENCE_NO) {
 			throw new IllegalArgumentException("Sequence number must have max 48 bits");
 		}
-		this.sequenceNumber = sequenceNumber;
-		if (session != null && session.getWriteState() != null && epoch > 0) {
-			fragmentBytes = encryptFragment(fragment.toByteArray());
+		if (this.sequenceNumber != sequenceNumber) {
+			this.sequenceNumber = sequenceNumber;
+			if (session != null && session.getWriteState() != null && epoch > 0) {
+				byte[] byteArray = fragment.toByteArray();
+				byteArray = applyInnerTypeAndPadding(byteArray);
+				fragmentLength = byteArray.length;
+				fragmentBytes = encryptFragment(byteArray);
```

Comment:
FMPOV, I don't like a setter, which then have something "always call additionally" in the javadoc.
Before the change,  encryptFragment was called, now with the new record type, the inner type and padding must also be applied. The grants before this change was, that the new sequence number also triggers a new encryption, because the explicit nonce is changed. Now the same grants as before are kept.

So should we go for rename the setter to `update`?

Reply:
>FMPOV, I don't like a setter, which then have something "always call additionally" in the javadoc.

It's not better I agree.

So you're right it's just a naming issue, `update` sounds good.
> But you didn't answer to the first question  : it seems pretty much the same idea that setFragment.
> I don't know if this is worthy to factorize this code ?

Reply:
> worthy to factorize this code

If I would guess, what you have in mind ...
Do you want the two 4-lines snippet to be merged into a single function?

Reply:
```
byte[] byteArray = fragment.toByteArray();
// the current length of the unprotected message
// this value is needed to generate the additional data when using AEAD
fragmentLength = byteArray.length;
```

Verifying the comment, I found that it's not (longer?) valid. I would prefer to clean it up with the commit of the wip adding more cipher suites.

Reply:
See e4df69e647bce

Reply:
I see, this would be changed in next PR